package.json

```diff
@@ -12,15 +12,23 @@
     "lint": "eslint --cache --cache-strategy content --report-unused-disable-directives .",
     "test": "cross-env NODE_ENV=test mocha --require ./mocha.bootstrap.js \"test/*.ts\"",
     "docs:dev": "docsify serve docs",
-    "migrate": "node bin/node-pg-migrate",
+    "migrate": "node bin/node-pg-migrate.js",
     "prepublishOnly": "pnpm run clean && pnpm install && pnpm run build",
     "preflight": "pnpm install && run-s format build lint test"
   },
   "bin": {
-    "node-pg-migrate": "bin/node-pg-migrate"
+    "node-pg-migrate": "bin/node-pg-migrate.js"
   },
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "require": "./dist/index.js",
+      "default": "./dist/index.js"
+    },
+    "./package.json": "./package.json"
+  },
```

Comment:
@bluwy could you check if this does not break anything to before? Or are there any hidden things that could bring users into conflict?

Reply:
To prevent any breaking changes, you'd have to do `"./*": "./*"` which will allow all the files like before. Otherwise this can only be released as a major (unless you're really confident that users aren't accessing any files they shouldn't be accessing)

I'd also remove the `"./package.json"` export. It isn't really needed anymore.

Reply:
> is `"./package.json"` not needed, because it would be covered by `"./*": "./*"`?


Partially yes, but the real reason is that `"./package.json"` used to be a convention in the past for toolings to read the `package.json`, but we don't have to follow the convention now. Any tooling that wants to be robust would've implemented a manual read anyway and can't rely on the convention. If you want to export `"./package.json"` still for convenience, it's certainly not wrong though.

> also, would it not be `"./*": "./dist/*"`?

No, its `"./*": "./*"`, at least for the non-breaking change standpoint. Otherwise if you access `node-pg-migrate/dist/migration-builder.js` for example, it maps to `node-pg-migrate/dist/dist/migration-builder.js`.

> is it previously possible to even access e.g. `require('node-pg-migrate/migration-builder')`, or would the devs need to enforce something like `require('node-pg-migrate/dist/migration-builder.js')`?

It's not possible to access that way, you'd need `node-pg-migrate/dist/migration-builder.js` instead. But seems like access dist [is quite common](https://github.com/search?q=node-pg-migrate%2Fdist&type=code) though.

Reply:
Oh wow 😮 that was VERY helpful and clarified so much, thx 🙏 

I think I will go directly with the breaking change anyway as I do already also drop node support <16 and so on 

But this will show up in the migration guide that I need to write!